package idgs.store.pb;

option java_outer_classname = "PbDataSync";

message Destroy {
}

message PartitionInfo {
  required int32 partition_id = 1;
}

message MigrationRequest {
  required int32 partition_id = 1;
  required string schema_name = 2;
  required string store_name  = 3;
}

message KeyValuePair {
  required string operation_name = 1;
  required bytes key             = 2;
  optional bytes value           = 3;
  optional uint64 version        = 4;
}

message MigrationData {
  repeated KeyValuePair data = 1;
}

message StoreMigrationComplete {
  required string schema_name = 1;
  required string store_name  = 2;
  optional uint64 data_size   = 3;
}

message PartitionMigrationComplete {
  required int32 partition_id = 1;
}

message SourceMemberLeaveEvent {
}

message CancelMigration {
  required int32 partition_id = 1;
  required string schema_name = 2;
  required string store_name  = 3;
}

message SyncRequest {
  required string schema_name = 1;
  required string store_name  = 2;
}

message SyncData {
  repeated KeyValuePair data = 1;
}

message SyncComplete {
  required string schema_name = 1;
  required string store_name  = 2;
  optional uint32 data_size   = 3;
}